modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/DisplayContextUtil.java

```diff
@@ -161,6 +162,59 @@ public static Map<Long, String> getTypeNames(
 		return typeNames;
 	}
 
+	public static List<String> getTypeNamesBySite(
+		long ctCollectionId, long groupId, boolean showHideable,
+		ThemeDisplay themeDisplay) {
+
+		List<String> typeNames = new ArrayList<>();
+
+		Searcher searcher = _searcherSnapshot.get();
+
+		SearchRequestBuilderFactory searchRequestBuilderFactory =
+			_searchRequestBuilderFactorySnapshot.get();
+
+		Sorts sorts = _sortsSnapshot.get();
+
+		SearchRequestBuilder searchRequestBuilder =
+			searchRequestBuilderFactory.builder(
+			).companyId(
+				themeDisplay.getCompanyId()
+			).entryClassNames(
+				CTEntry.class.getName()
+			).emptySearchEnabled(
+				true
+			).fields(
+				"modelClassNameId", "typeName", Field.GROUP_ID
+			).sorts(
+				sorts.field(
+					Field.getSortableFieldName(
+						"typeName_".concat(
+							LocaleUtil.toLanguageId(themeDisplay.getLocale()))),
+					SortOrder.ASC)
+			).withSearchContext(
+				searchContext -> {
+					searchContext.setAttribute(
+						"ctCollectionId", ctCollectionId);
+					searchContext.setAttribute("showHideable", showHideable);
+				}
+			);
+
+		SearchResponse searchResponse = searcher.search(
+			searchRequestBuilder.build());
+
+		for (Document document : searchResponse.getDocuments()) {
+			if (document.getLong(Field.GROUP_ID) == null) {
+				continue;
+			}
+
+			if (document.getLong(Field.GROUP_ID) == groupId) {
+				typeNames.add(document.getString("typeName"));
+			}
```

Comment:
I remember trying to add `searchContext.setAttribute("groupId", groupId);` but it would return all entries for both sites 
<img width="915" alt="group" src="https://github.com/user-attachments/assets/7f03bf77-7ad8-4b86-8acb-801aab07db12">

Reply:
The filter expects the `groupId` attribute to be an array so you'll need to pass the value as an array. See https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/search/spi/model/query/contributor/CTEntryModelPreFilterContributor.java#L46-L47